pkgs/by-name/up/updog/package.nix

```
@@ -1,35 +1,40 @@
 {
   lib,
   python3Packages,
-  fetchPypi,
+  fetchFromGitHub,
+  versionCheckHook,
 }:
 
 python3Packages.buildPythonApplication rec {
   pname = "updog";
   version = "1.4";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "7n/ddjF6eJklo+T79+/zBxSHryebc2W9gxwxsb2BbF4=";
+  src = fetchFromGitHub {
+    owner = "sc0tfree";
+    repo = "updog";
+    tag = version;
+    hash = "sha256-e6J4Cbe9ZRb+nDMi6uxwP2ZggbNDyKysQC+IcKCDtIw=";
   };
 
-  propagatedBuildInputs = with python3Packages; [
+  dependencies = with python3Packages; [
     colorama
     flask
     flask-httpauth
     werkzeug
     pyopenssl
   ];
 
-  checkPhase = ''
-    $out/bin/updog --help > /dev/null
-  '';
+  nativeCheckInputs = [ versionCheckHook ];
+  versionCheckProgramArg = "--version";
 
-  meta = with lib; {
+  # no python tests
+
+  meta = {
     description = "Updog is a replacement for Python's SimpleHTTPServer";
```

```suggestion
    description = "Replacement for Python's SimpleHTTPServer";
```

The description shouldn't really contain the name of the package.

---

Will fix. Thanks